### PR TITLE
Feature/jsdoc

### PR DESCRIPTION
**Pull request by chrispascucci**
NOTES: 
* This branch needs additional documentation, but it contains the initial setup for jsdoc.
* The official JSDoc spec does not yet support ES6 Promises.  More detail can be found here: https://github.com/jsdoc3/jsdoc/issues/1197  Apparently the decision about the syntax has been made (as of 11/2015), but not yet implemented (as of 2/2017).
* I added a prestart command to the package.json file to generate the jsdoc UI and statically defined the route to it in app.js.  I'd like to modify this so that it works with the existing hot-reloading setup.  Currently the UI is not regenerated automatically when changes are made to files; one must manually stop/start the api.
* Because JSDoc incorporates the readme file into it's UI, I used /api-docs/ as the route for JSDoc and modified the Swagger route to be /api-docs/swagger, a link to which can be found in the readme.
* The UI is useable, but the templates should be reviewed by someone from the Digital team to bring them in line with proper Fusion branding/layouts.

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -46,7 +46,8 @@ app.get('/swagger.json', function(request, response) {
   response.json(swaggerSpec);
 });
 
-app.use('/api-docs', swaggerUi.serve, swaggerUi.setup(swaggerSpec));
+app.use('/api-docs/swagger', swaggerUi.serve, swaggerUi.setup(swaggerSpec));
+app.use('/api-docs', express.static('jsdoc/'));
```
Comment (rclanan) [inline server/app.js]: According to issue #14 
"Output of this should be API documentation for API developers, available at the following route: api/dev-docs."

Comment (rclanan) [inline server/app.js]: Issues #18 
"This information will be available via an API call; i.e., a GET request to https://theHost.com/api-docs will return information similar to that shown in the API_doc_example.md document."

Comment (BWortman) [inline server/app.js]: Yes, I'd rather not have "swagger" in the path... want to remain neutral as to the implementation. But we don't have a path designed for this sort of "internal" documentation. How about leaving the swagger the way it was (/api-docs), and making this JSDoc available at /dev-docs? The first path documents the API, and the second one is documentation for "internal" developers.

Comment (rclanan) [inline server/app.js]: We can make whatever paths we want. 

/doc/api/
/doc/dev/
Maybe?

Comment (BWortman) [inline server/app.js]: Yep, that'll work.

File: package.json [hunk 2]
```diff
@@ -7,6 +7,8 @@
   "description": "Node.js + Express implementation of the RESTful AutoRenter API.",
   "main": "server/index.js",
   "scripts": {
+    "jsdoc": "jsdoc --configure .jsdocconf",
+    "prestart": "npm run jsdoc",
```
Comment (BWortman) [inline package.json]: Let's instead add the command to the travis file so this doesn't happen every build.

File: package.json [hunk 3]
```diff
@@ -7,6 +7,8 @@
   "description": "Node.js + Express implementation of the RESTful AutoRenter API.",
   "main": "server/index.js",
   "scripts": {
+    "jsdoc": "jsdoc --configure .jsdocconf",
```
Comment (BWortman) [inline package.json]: Please verify that the output folder is cleaned on build so that obsolete files don't accumulate.

File: .travis.yml [hunk 4]
```diff
@@ -9,4 +9,6 @@ notifications:
     on_success: change
     on_failure: always
     on_start: never
-after_script: "cat ./coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js"
+after_script: 
+      - "cat ./coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js"
+      - "npm run jsdoc"
```
Comment (BWortman) [inline .travis.yml]: Do we need both this AND the prestart? Local devs can manually build whenever they want. We need travis to do the doc build. But I was thinking we were going to eliminate the prestart.

File: package.json [hunk 5]
```diff
@@ -7,6 +7,8 @@
   "description": "Node.js + Express implementation of the RESTful AutoRenter API.",
   "main": "server/index.js",
   "scripts": {
+    "jsdoc": "rm -rf ./jsdoc && jsdoc --configure .jsdocconf",
+    "prestart": "npm run jsdoc",
```
Comment (BWortman) [inline package.json]: I think we agreed to remove this, provided travis was properly integrated to build the docs.

File: .travis.yml (hunk 4 above)
Comment (BWortman) [inline .travis.yml]: I removed the prestart, which may result in us having no jsdoc in the "production" build if we got the travis config wrong. I'm also not sure what the impact will be on the heroku build. Guess we have to merge to find out...?

**Comment by JARWATI** [comment]
Yes on pull request

**Pull request merged**